src/dao/V2FilterProcessor.ts

```diff
@@ -0,0 +1,117 @@
+import {IHash} from '../internal/IHash';
+
+import {IFilterProcessor} from '../api/IFilterProcessor';
+
+import {Filter} from '../api/Filter';
+import {Comparator, Comparators} from '../api/Comparator';
+import {Restriction} from '../api/Restriction';
+import {NestedRestriction} from '../api/NestedRestriction';
+import {OnmsError} from '../api/OnmsError';
+import {Operator, Operators} from '../api/Operator';
+import {Clause} from '../api/Clause';
+
+/**
+ * OpenNMS V2 ReST filter processor
+ * @module V2FilterProcessor
+ */ /** */
+export class V2FilterProcessor implements IFilterProcessor {
+
+  /** constant used to represent null values in the V2 API */
+  public static NULL_VALUE = '\u0000';
+
+  /** constant used to represent null dates in the V2 API
+   *  this must be explicitly set as the restriction value when using
+   *  either the NULL or NOTNULL comparators on date fields
+   */
+  public static NULL_DATE = '1970-01-01T00:00:00.000+0000';
+
+  /**
+   * given a comparator, convert it to a correspond comparator
+   * that can be used in the FIQL expression
+   */
+  private static toFIQLComparator(comparator: Comparator) {
+    switch (comparator) {
+      case Comparators.EQ:
+      case Comparators.NULL:
+        return '==';
+      case Comparators.NE:
+      case Comparators.NOTNULL:
+        return '!=';
+      case Comparators.GT:
+        return '=gt=';
+      case Comparators.LT:
+        return '=lt=';
+      case Comparators.GE:
+        return '=ge=';
+      case Comparators.LE:
+        return '=le=';
+      case Comparators.LIKE:
+        return '==';
+      case Comparators.ILIKE:
+      default:
+        throw new OnmsError('Unsupported comparator type: ' + comparator);
+    }
+  }
+
+  /** given a restriction, compute the value to use in the FIQL expression */
+  private static toFIQLValue(restriction: Restriction) {
+    switch (restriction.comparator) {
+      case Comparators.NULL:
+      case Comparators.NOTNULL:
+        return restriction.value === undefined ? V2FilterProcessor.NULL_VALUE : restriction.value;
+      case Comparators.LIKE:
+        return '*' + restriction.value + '*';
```

Comment:
The v2 API is automatically applying the like comparator if the value contains a wildcard.
See https://github.com/OpenNMS/opennms/blob/c537bf85351f4248d25a41333c06b5d75b3cf5f8/opennms-webapp-rest/src/main/java/org/opennms/web/rest/support/CriteriaBuilderSearchVisitor.java#L97-L97 for more details. Explicitly setting it may also work, but from my point of view allowing the definition of `someAttribute = *someValue*` and applying the like comparator automatically is probably the better user experience. Especially as the value could be something like `my*value`.